tests/callbacks/test_callbacks.py

```diff
@@ -45,6 +47,33 @@ def clean_mlflow_runs():
         yield
 
 
+@contextlib.contextmanager
+def maybe_patch_mlflow_for_trash(cb_cls: type[Callback]) -> Generator[None, None, None]:
```

Comment:
this is the one i said you should add to get_cb_patches, so it only needs to be run for mlflow logger